corehq/motech/repeaters/static/repeaters/js/bootstrap3/repeat_record_report.js

```diff
@@ -156,49 +156,47 @@ hqDefine('repeaters/js/bootstrap3/repeat_record_report', function () {
         });
 
         function performAction(action) {
-            if (isAnythingChecked()) {
+            var bulkSelection = bulkSelectionChecked();
+            if (bulkSelection || areRecordsChecked()) {
+                if (bulkSelection) { setFlag(bulkSelection); }
+                // only applies to checked items, not bulk selections
+                // leaving as is to preserve behavior
                 if (isActionPossibleForCheckedItems(action)) {
-                    $('#warning').addClass('hide');
-                    $('#not-allowed').addClass('hide');
+                    hideAllWarnings();
                     $popUp.modal('show');
                 } else {
-                    $('#warning').addClass('hide');
-                    $('#not-allowed').removeClass('hide');
+                    showWarning('not-allowed');
                 }
             } else {
-                $('#warning').removeClass('hide');
-                $('#not-allowed').addClass('hide');
+                showWarning('no-selection');
             }
         }
 
-        function isAnythingChecked() {
-            var items = document.getElementsByName('xform_ids');
-
+        function bulkSelectionChecked() {
             if (selectAll.checked) {
-                setFlag('select_all');
-                return true;
+                return 'select_all';
             } else if (selectPending.checked) {
-                setFlag('select_pending');
-                return true;
+                return 'select_pending';
             } else if (selectCancelled.checked) {
-                setFlag('select_cancelled');
-                return true;
+                return 'select_cancelled';
             }
+        }
 
-            for (var i = 0; i < items.length; i++) {
-                if (items[i].checked) {
+        function areRecordsChecked() {
+            var items = document.getElementsByName('xform_ids');
```

Comment:
Nit: `find` would be nice and concise here.

Reply:
Not this exactly but in a similar vein, I ended up restructuring so this is `getCheckedRecords`. https://github.com/dimagi/commcare-hq/pull/35047/commits/cdaaaedfb586863a9047340af3232a5400fbc0ef